src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/controllers/web/ReportControllerTests.kt

```diff
@@ -108,6 +108,110 @@ class ReportControllerTests : TeamcityTests()
     }
 
     @Test
+    fun `artefacts have expected urls`()
+    {
+        val artefacts = listOf(Artefact(ArtefactFormat.DATA, "desc", listOf("unsuitable.csv")),
+                Artefact(ArtefactFormat.DATA, "desc", listOf("another.csv", "suitable.png")))
+
+        val orderly = mock<OrderlyClient> {
+            on { this.getDetailsByNameAndVersion("r1", "v1") } doReturn
+                    mockReportDetails.copy(artefacts = artefacts)
+        }
+
+        val sut = ReportController(actionContext, orderly)
+        val result = sut.getByNameAndVersion()
+
+        assertThat(result.artefacts.count()).isEqualTo(2)
+
+        assertThat(result.artefacts[0].artefact).isEqualTo(artefacts[0])
+        assertThat(result.artefacts[0].files.count()).isEqualTo(1)
+        assertThat(result.artefacts[0].files[0].name).isEqualTo("unsuitable.csv")
+        assertThat(result.artefacts[0].files[0].url).isEqualTo("/reports/r1/versions/v1/artefacts/unsuitable.csv")
```

Comment:
Do mean a selenium test to check that you can actually log in with a browser and download files? I'm already putting in integration tests like others we have to check if the endpoint is secured (your other suggestion), but actually checking for a 200 download through integration tests is tricky because the web endpoints are secured with the indirect clients.

Reply:
Hm, actually I don't think those integration tests are doing quite what I expected anyway. Let's talk about this when you're in!